### PR TITLE
chain: Refactor SendOracleResponse out and improve request ibc flow

### PR DESCRIPTION
**Pull request by sorawit**
- Make SendOracleResponse not depend on request keeper at all. It just sends packets.
- Make sure expired requests get sent back to requester.
- Use error as result for failed requests.
- Merge NewRequest function into one.

### CODE REVIEW

File: chain/x/oracle/keeper/request.go [hunk 1]
```diff
@@ -78,110 +76,60 @@ func (k Keeper) AddRequest(ctx sdk.Context, req types.Request) (types.RequestID,
 	return id, nil
 }
 
-func (k Keeper) resolveRequest(
-	ctx sdk.Context, reqID types.RequestID, resolveStatus types.ResolveStatus, result []byte,
+func (k Keeper) ResolveRequest(
```
Comment (taobun) [inline chain/x/oracle/keeper/request.go:81]: Can you add a comment to make this function more understandable?

File: chain/x/oracle/endblock.go [hunk 2]
```diff
@@ -14,13 +14,16 @@ func resolveRequest(ctx sdk.Context, k Keeper, reqID types.RequestID) {
 	script := k.MustGetOracleScript(ctx, req.OracleScriptID)
 	executeGas := k.GetParam(ctx, KeyExecuteGas)
 	result, _, err := k.OwasmExecute(env, script.Code, "execute", req.Calldata, executeGas)
-	var resolveStatus types.ResolveStatus
+
+	var res types.OracleResponsePacketData
 	if err != nil {
-		resolveStatus = types.Failure
+		res = k.ResolveRequest(ctx, reqID, types.Failure, []byte(err.Error()))
```
Comment (taobun) [inline chain/x/oracle/endblock.go]: Error has been used?

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/endblock.go (hunk 2 above)
Comment (sorawit) [reply chain/x/oracle/endblock.go]: Actually error should not be part of the consensus. I do want to allow Owasm to emit errors somehow though, but will save that for other diffs.

File: chain/x/oracle/keeper/request.go (hunk 1 above)
Comment (sorawit) [reply chain/x/oracle/keeper/request.go:81]: Yes can do.

**Review by taobun — APPROVED** [review]

**Review by Benzbeeb — APPROVED** [review]

**Pull request merged**